samza-core/src/main/java/org/apache/samza/clustermanager/container/placement/ContainerPlacementRequestAllocator.java

```diff
@@ -0,0 +1,84 @@
+ * under the License.
+ */
+package org.apache.samza.clustermanager.container.placement;
+
+import com.google.common.base.Preconditions;
+import org.apache.samza.clustermanager.ContainerProcessManager;
+import org.apache.samza.container.placement.ContainerPlacementRequestMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Stateless handler that periodically dispatches {@link ContainerPlacementRequestMessage} read from Metadata store to Job Coordinator
+ */
+public class ContainerPlacementRequestAllocator implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ContainerPlacementRequestAllocator.class);
+  private static final int DEFAULT_CLUSTER_MANAGER_CONTAINER_PLACEMENT_HANDLER_SLEEP_MS = 5000;
+
+  /**
+   * {@link ContainerProcessManager} needs to intercept container placement actions between ContainerPlacementRequestAllocator and
+   * {@link org.apache.samza.clustermanager.ContainerManager} to avoid cyclic dependency between
+   * {@link org.apache.samza.clustermanager.ContainerManager} and {@link org.apache.samza.clustermanager.ContainerAllocator}
+   */
+  private final ContainerProcessManager containerProcessManager;
+  private final ContainerPlacementMetadataStore containerPlacementMetadataStore;
+  /**
+   * State that controls the lifecycle of the ContainerPlacementRequestAllocator thread
+   */
+  private volatile boolean isRunning;
+
+  public ContainerPlacementRequestAllocator(ContainerPlacementMetadataStore containerPlacementMetadataStore, ContainerProcessManager manager) {
+    Preconditions.checkNotNull(containerPlacementMetadataStore, "containerPlacementMetadataStore cannot be null");
+    Preconditions.checkNotNull(manager, "ContainerProcessManager cannot be null");
+    this.containerProcessManager = manager;
+    this.containerPlacementMetadataStore = containerPlacementMetadataStore;
+    this.isRunning = true;
+  }
+
+  @Override
+  public void run() {
+    while (isRunning && containerPlacementMetadataStore.isRunning()) {
+      try {
+        for (ContainerPlacementRequestMessage message : containerPlacementMetadataStore.readAllContainerPlacementRequestMessages()) {
+          // We do not need to dispatch ContainerPlacementResponseMessage because they are written from JobCoordinator
+          // in response to a Container Placement Action
+          LOG.info("Received a container placement message {}", message);
+          containerProcessManager.registerContainerPlacementAction(message);
+        }
+        Thread.sleep(DEFAULT_CLUSTER_MANAGER_CONTAINER_PLACEMENT_HANDLER_SLEEP_MS);
+      } catch (InterruptedException e) {
+        LOG.warn("Got InterruptedException in ContainerPlacementRequestAllocator thread.", e);
+        Thread.currentThread().interrupt();
+      } catch (Exception e) {
+        LOG.error(
+            "Got unknown Exception while reading ContainerPlacementRequestMessage in ContainerPlacementRequestAllocator thread",
```

Comment:
Well, if we know the Exception so its not really "unknown" :-)

Reply:
removed "unknown"